library/src/scala/quoted/Quotes.scala

```diff
@@ -4151,9 +4151,19 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
     /** Extension methods of `SourceFile` */
     trait SourceFileMethods {
       extension (self: SourceFile)
-        /** Path to this source file */
+        /** Path to this source file. May be `null` for virtual files such as in the REPL.  */
+        @deprecated("Use getJPath, name, or path instead of jpath", "3.0.2")
         def jpath: java.nio.file.Path
 
+        /** Path to this source file. May be `None` for virtual files such as in the REPL. */
+        def getJPath: Option[java.nio.file.Path]
+
+        /** Name of the source file */
+        def name: String
+
+        /** Path of the source file */
```

Comment:
I think the comment should mention that this isn't necessarily a path to a file on disk because of virtual files, otherwise people might try to convert it to a java Path/File.